src/core/document/json.pl

```diff
@@ -2912,10 +2995,12 @@
         get_dict('@id', Document, Id),
         error(missing_field('@id', Document), _)),
     check_json_string('@id', Id),
+    database_prefixes(Transaction, Prefixes),
     database_schema(Transaction, Schema),
-
-    database_prefixes(Transaction, Context),
-    prefix_expand_schema(Id,Context,Id_Ex),
+    prefix_expand_schema(Id,Prefixes,Id_Ex),
+    do_or_die(
+        valid_schema_name(Prefixes,Id_Ex),
+        error(can_not_insert_class_with_reserve_name(Id), _)),
```

Comment:
This error should be caught, a useful error message should be printed, and a test should be added for that message.